src/System.Linq.Expressions/src/System/Linq/Expressions/BinaryExpression.cs

```diff
@@ -893,8 +893,7 @@ private static void ValidateUserDefinedConditionalLogicOperator(ExpressionType n
         private static void VerifyOpTrueFalse(ExpressionType nodeType, Type left, MethodInfo opTrue, string paramName)
         {
             ParameterInfo[] pmsOpTrue = opTrue.GetParametersCached();
-            if (pmsOpTrue.Length != 1)
-                throw Error.IncorrectNumberOfMethodCallArguments(opTrue, paramName);
+            Debug.Assert(pmsOpTrue.Length == 1);
```

Comment:
I realize it's dead code, but I think this should be reverted.  The entire purpose of this VerifyOpTrueFalse method is to verify and throw if the provided member is not true/false, and that includes validating the number of arguments.